packages/components/cascader/src/cascader.ts

```diff
@@ -0,0 +1,83 @@
+import { CommonProps } from '@element-plus/components/cascader-panel'
+import {
+  buildProps,
+  isBoolean,
+  isValidComponentSize,
+} from '@element-plus/utils'
+import { useTooltipContentProps } from '@element-plus/components/tooltip'
+import { tagProps } from '@element-plus/components/tag'
+import { CHANGE_EVENT, UPDATE_MODEL_EVENT } from '@element-plus/constants'
+import type { ComponentSize } from '@element-plus/constants'
+import type { PropType } from 'vue'
+import type {
+  CascaderNode,
+  CascaderValue,
+} from '@element-plus/components/cascader-panel'
+
+export const cascaderProps = buildProps({
+  ...CommonProps,
+  size: {
+    type: String as PropType<ComponentSize>,
+    validator: isValidComponentSize,
+  },
+  placeholder: {
+    type: String,
+  },
+  disabled: Boolean,
+  clearable: Boolean,
+  filterable: Boolean,
+  filterMethod: {
+    type: Function as PropType<
```

Comment:
Maybe use `definePropType`, and the same in other places.

Reply:
done.